hooks/tk-multi-publish2/basic/collector.py

```diff
@@ -0,0 +1,244 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import glob
+import os
+import maya.cmds as cmds
+import maya.mel as mel
+import sgtk
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class MayaSessionCollector(HookBaseClass):
+    """
+    Collector that operates on the maya session. Should inherit from the basic
+    collector hook.
+    """
+
+    def process_current_session(self, parent_item):
+        """
+        Analyzes the current session open in Maya and parents a subtree of
+        items under the parent_item passed in.
+
+        :param parent_item: Root item instance
+        """
+
+        # create an item representing the current maya session
+        item = self.collect_current_maya_session(parent_item)
+        project_root = item.properties["project_root"]
+
+        # look at the render layers to find rendered images on disk
+        self.collect_rendered_images(item)
+
+        # if we can determine a project root, collect other files to publish
+        if project_root:
+
+            self.logger.info(
+                "Current Maya project is: %s." % (project_root,),
+                extra={
+                    "action_button": {
+                        "label": "Change Project",
+                        "tooltip": "Change to a different Maya project",
+                        "callback": lambda: mel.eval('setProject ""')
+                    }
+                }
+            )
+
+            self.collect_playblasts(item, project_root)
+            self.collect_alembic_caches(item, project_root)
+
+        else:
+
+            self.logger.warning(
+                "Could not determine the current Maya project.",
```

Comment:
i don't think a lot of medium size studios use maya projects. I am not sure about the relationship between maya projects and templates - for zero configs, we used the maya project as (an initial) way to go look for things like renders, playblasts etc and present to the user.

I am not sure what has been discussed in terms of the workflow for the classic setup - in my mind, when you operate with zero config, you want to collect all items you can find to the best of our ability without knowing anything except the maya file path. With classic, we use templates instead to define our naming convention and i would have guessed we wanted to use templates to define places where the collector would go look for things - like files, playblasts, renders etc. Adding a new template adds a new 'standard location' that the collector will sniff out.

Reply:
it might be worth working through some user stories to make sure we are hitting all the right 'classic' notes here - we should at least be able to hit the same use cases as the old publisher right (which didn't rely on maya projects). Happy to discuss this further!

Reply:
+1 on what @manneohrstrom is saying.  Josh and I did talk about that already and it is on radar, just not in there yet.

Reply:
@gomezll is working on a publish plugin to export and publish an alembic cache for the scene geometry (same thing publish1 does). That's a first pass at validating the "export & publish" setup. We can add something similar for cameras or whatever else we want to do once we have that and it feels like it's in a good spot. I'm also thinking about a setting on the maya collector whereby you could simply configure a list of templates to search for. Any matched paths would be handed to the base file publisher just like they were dragged/dropped. That seems pretty useful for the "already exported + publish" workflow.